src/npctalk.cpp

```diff
@@ -2316,6 +2316,33 @@ void talk_effect_fun_t::set_queue_effect_on_condition( const JsonObject &jo,
         }
     };
 }
+
+void talk_effect_fun_t::set_weighted_list_eocs( const JsonObject &jo,
+        const std::string &member )
+{
+    weighted_int_list<effect_on_condition_id> eocs;
+    for( JsonArray pair : jo.get_array( member ) ) {
+        effect_on_condition_id eoc;
+        int weight = 1;
+        for( JsonValue jv : pair ) {
+            if( jv.test_int() ) {
+                weight = jv.get_int();
+            } else {
+                eoc = effect_on_conditions::load_inline_eoc( jv, "" );
+            }
+        }
+        eocs.add( eoc, pair.get_int( 1 ) );
```

Comment:
Do you need `weight` variable?

Reply:
I'm not sure what you mean exactly, I could add one to be a little clearer but I don't think I need one?